src/Controls/src/Core/RowDefinitionCollectionTypeConverter.cs

```diff
@@ -24,11 +26,8 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
 			if (strValue != null)
 			{
 				var lengths = strValue.Split(',');
-				var coldefs = new RowDefinitionCollection();
 				var converter = new GridLengthTypeConverter();
-				foreach (var length in lengths)
-					coldefs.Add(new RowDefinition { Height = (GridLength)converter.ConvertFromInvariantString(length) });
-				return coldefs;
+				return new RowDefinitionCollection(lengths.Select(length => new RowDefinition {Height = (GridLength)converter.ConvertFromInvariantString(length) }).ToArray());
```

Comment:
I would suggest not to use Linq Select and ToArray here and just use the foreach as before. The performance overhead isn't worth it. We spend a decent amount of time removing Linq usages in dotnet/runtime.